src/Importer/Importer.php

```diff
@@ -236,6 +237,9 @@ protected function executeImport(array $items): bool
         }
 
         try {
+            $event = $this->eventDispatcher->dispatch(BeforeImportItemsEvent::NAME, new BeforeImportItemsEvent($items, $this->configModel, $this->source, $this->dryRun));
+            $items = $event->getItems();
```

Comment:
Wouldn't that be redudant? We already have the BeforeImportEvent which is thrown after the source has been retrieved? You would also be able to change the items here as in the event you propose.

Reply:
I set the event there to be able to throw the exception were it needs to be. But I could set an error flag on the source in the BeforeImportEvent as well and check for that. Would that be a better solution?